### PR TITLE
HDDS-4721. Allow Default Space Quota Unit as Bytes

### PR DESCRIPTION
**Pull request by amaliujia**
## What changes were proposed in this pull request?

Allow Default Space Quota Unit as Bytes.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4721

## How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @captainzmc

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java [hunk 1]
```diff
@@ -704,6 +704,16 @@ public void testShQuota() throws Exception {
...
+    // test whether supports default quota unit as bytes.
+    String[] bucketArgs5 = new String[]{"bucket", "setquota", "vol4/buck4",
+        "--space-quota", "500"};
+    execute(ozoneShell, bucketArgs5);
+    out.reset();
+    assertEquals(500, objectStore.getVolume("vol4")
+        .getBucket("buck4").getQuotaInBytes());
+    assertEquals(100, objectStore.getVolume("vol4")
+        .getBucket("buck4").getQuotaInNamespace());
+
```
Comment (ayushtkn) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java:716]: Isn't this same as L704? If so, I don't think we changed anything on the namespace quota side, so no need of this?

Comment (amaliujia) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java:716]: O yes. That's more like a extra check to see the namespace quota remains the same. But removing it won't hurt anything.

Comment (amaliujia) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java]: Removed the namespace check as it is not relevant

**Comment by captainzmc** [comment]
Thanks for @amaliujia‘s PR.  LGTM overall. 
Can you also add test case in TestOzoneRpcClientAbstract.testSetAndClrQuota() for test Java API?

**Comment by amaliujia** [comment]
@captainzmc done!

**Review by captainzmc — APPROVED** [review]
LGTM +1
If there is no new comment, let's merge the PR tomorrow.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java [hunk 2]
```diff
@@ -193,6 +193,11 @@ public static OzoneQuota parseSpaceQuota(String quotaInBytes) {
           break;
         }
       }
+      // there might be no unit specified.
+      if (size.equals("")) {
+        size = uppercase;
+        currUnit = Units.B;
```
Comment (ayushtkn) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java:199]: if size is empty, currUnit will be Units.B already, do we need to re assign it here

Comment (amaliujia) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java]: Done!

**Review by ayushtkn — APPROVED** [review]
LGTM +1

**Comment by ayushtkn** [comment]
Thanx @amaliujia for the contribution and @captainzmc for the review!!!

**Pull request merged**